### PR TITLE
Allow tools to use service account credentials

### PR DESCRIPTION
**Pull request by cdonati**

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1135/

File: lib/agents/gce_agent.py [hunk 1]
```diff
@@ -1064,6 +1070,24 @@ def cleanup_state(self, parameters):
...
+    Args:
+      secrets_location: A string that contains the location of the JSON
+        credentials file downloaded from GCP.
+    Returns:
+      A string containing the type of credentials to use.
+    """
+    with open(secrets_location) as secrets_file:
+      secrets_json = secrets_file.read()
+    secrets = json.loads(secrets_json)
+    if 'type' in secrets and secrets['type'] == CredentialTypes.SERVICE:
```
Comment (menivaitsi) [inline lib/agents/gce_agent.py]: Can 'type' be a constant?

File: lib/agents/gce_agent.py [hunk 2]
```diff
@@ -1080,9 +1104,16 @@ def open_connection(self, parameters):
     # Perform OAuth 2.0 authorization.
     flow = None
     if self.PARAM_SECRETS in parameters:
-      flow = oauth2client.client.flow_from_clientsecrets(
-        os.path.expanduser(parameters[self.PARAM_SECRETS]), 
-        scope=self.GCE_SCOPE)
+      secrets_location = os.path.expanduser(parameters[self.PARAM_SECRETS])
+      secrets_type = GCEAgent.get_secrets_type(secrets_location)
+      if secrets_type == CredentialTypes.SERVICE:
+        scopes = ['https://www.googleapis.com/auth/compute']
```
Comment (menivaitsi) [inline lib/agents/gce_agent.py]: Please turn the scopes list into a constant at the top.

File: lib/remote_helper.py [hunk 3]
```diff
@@ -638,14 +640,15 @@ def copy_deployment_credentials(cls, host, options):
...
-          options.verbose)
-      cls.scp(host, options.keyname, LocalState.get_oauth2_storage_location(
-        options.keyname), '/etc/appscale/oauth2.dat', options.verbose)
-
+      secrets_location = LocalState.get_client_secrets_location(options.keyname)
+      if not os.path.exists(secrets_location):
+        raise AppScaleException('{} does not exist.'.format(secrets_location))
+      secrets_type = GCEAgent.get_secrets_type(secrets_location)
+      cls.scp(host, options.keyname, secrets_location,
+        '/etc/appscale/client_secrets.json', options.verbose)
```
Comment (menivaitsi) [inline lib/remote_helper.py]: Can we pull /etc/appscale from another module?
If it's not defined as a constant somewhere, maybe we can define it in lib/remote_helper?

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1145/

**Pull request merged**